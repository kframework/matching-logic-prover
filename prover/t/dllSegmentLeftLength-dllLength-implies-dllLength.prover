claim \implies( \and( dllSegmentLeftLength ( 
                            variable("H") { ArrayIntInt }
                          , variable("X") { Int }
                          , variable("Y") { Int }
                          , variable("F") { SetInt }
                          , variable("L") { Int }
                          , .Patterns
                          )
                    , dllLength 
                          ( variable("H") { ArrayIntInt }
                          , variable("Y") { Int }
                          , variable("G") { SetInt }
                          , variable("M") { Int }
                          , .Patterns
                          )
                    , \equals( variable("K") { SetInt }
                             , union( variable("F") { SetInt }
                                    , variable("G") { SetInt }
                                    )
                             )
                    , \equals( variable("N") { Int }
                             , plus(variable("L") { Int }, variable("M") { Int }))
                    , disjoint( variable("F") { SetInt }
                              , variable("G") { SetInt }
                              )
                    , .Patterns
                    )
              , \and( dllLength (
                            variable("H") { ArrayIntInt }
                          , variable("X") { Int }
                          , variable("K") { SetInt }
                          , variable("N") { Int }
                          , .Patterns
                          )
                    , .Patterns
                    )
              )

strategy search-bound(3)


/*
strategy simplify ; kt 
       ; simplify 
       ; ( direct-proof  
         | right-unfold-Nth(0,1) ; simplify ; direct-proof 
         )
*/
