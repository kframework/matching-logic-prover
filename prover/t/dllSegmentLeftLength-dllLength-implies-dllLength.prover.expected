<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( dllSegmentLeftLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "F" ) { SetInt } , variable ( "L" ) { Int } , .Patterns ) , dllLength ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" ) { Int } , variable ( "G" ) { SetInt } , variable ( "M" ) { Int } , .Patterns ) , \equals ( variable ( "K" ) { SetInt } , union ( variable ( "F" ) { SetInt } , variable ( "G" ) { SetInt } ) ) , \equals ( variable ( "N" ) { Int } , plus ( variable ( "L" ) { Int } , variable ( "M" ) { Int } ) ) , disjoint ( variable ( "F" ) { SetInt } , variable ( "G" ) { SetInt } ) , .Patterns ) , \and ( dllLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "K" ) { SetInt } , variable ( "N" ) { Int } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
